Sources/XiEditor/EditView.swift

```diff
@@ -414,6 +414,24 @@ final class EditView: NSView, NSTextInputClient, TextPlaneDelegate {
         }
     }
 
+    func annotationColor(for annotation: Annotation) -> UInt32 {
+        let selectionColor = self.isFrontmostView ? dataSource.theme.selection : dataSource.theme.inactiveSelection ?? dataSource.theme.selection
+        let selArgb = colorToArgb(selectionColor)
+        let highlightColors = dataSource.theme.findHighlights
+        let highlightsArgb = (highlightColors ?? [])!.map({
```

Comment:
this looks weird, trying to get my head around it. If `highlightColors` can be `nil`, I think it makes sense to do the `?? []` when we declare it above here? But this is confusing, because if it can be nil and this can be empty then below we will be indexing into an empty array, which should crash?

Reply:
Right, I think it makes sense to use the selection color as default then if `highlightColors` is `nil`.

Reply:
this also might explain a bug I saw testing here where with the github theme the selection color was bright red?

Reply:
The selection color? Currently, the find highlight color for the first query is red but that shouldn't influence the selection color.